File: Kata/CheckOutApp/PizzaWahala.java
import java.util.Scanner;
import java.lang.Math;

			public class PizzaWahala{
				public static void main(String[] args){
					
				System.out.print("""
				
				Welcome to Iya Afeez Pizza Joint, Ajegunle.

							MENU  						
.........................................................................................................................................................
		Pizza type			Number of Slices		Price of box		
.........................................................................................................................................................

		1. Sapa Size			        	4			                 2,000			
		
		2. Small Money			   	   6			               	 2,400			
		
		3. Big Boys				      	8				         3,000			
	
		4. 0dogwu 				      	12				         5,200			
..........................................................................................................................................................	
		0: Exit

	
	What pizza type are you buying today?

	(pick options 1-4).
			""");

				Scanner input = new Scanner(System.in);
				int typeChosen = input.nextInt();

				switch(typeChosen){
			

				case 1: 
					System.out.print("How many customer are paying ?  ");	
						int customersForSapaSize  = input.nextInt();
						System.out.println(" ");

						if(customersForSapaSize > 0){
							int sapaSize = 4;
							int sapaPrice = 2000;
						
							double sapa = (double)customersForSapaSize / sapaSize;
							double sapaBoxCounter = Math.ceil(sapa);
							int sapaBoxes = (int) sapaBoxCounter;

							int sapaSlicesLeft = (sapaBoxes*sapaSize) - customersForSapaSize;
	

							float price = sapaPrice * sapaBoxes;

							System.out.println("          PURCHASE DETAILS");
							System.out.println("Pizza Type: sapa Size:");
							System.out.println(" ");

							System.out.println("  Total  Boxes: "+ sapaBoxes);
							System.out.println("	For "+ customersForSapaSize + "person(s)\n Total Price: #" + price + "\n LeftOver Slices:" +sapaSlicesLeft);
							System.out.println(" ");


							System.out.print("     Enjoy Your Pizza");

					}else{
						System.out.print("Invalid Input, please try again");
						}



						break;
		
				
				case 2:
					System.out.print("How many customers are paying ? ");
						int customersForsmallBoysSize = input.nextInt();
				
					if(customersForsmallBoysSize > 0){
						int smallBoysSize = 6;
						int smallBoysSizePrice = 2400;

						double smallBoys = (double)customersForsmallBoysSize / smallBoysSize;
						double smallBoysBoxCounter = Math.ceil(smallBoys);
						int smallBoysBoxes = (int) smallBoysBoxCounter;

						int smallBoysSlicesLeft = (smallBoysBoxes*smallBoysSize) - customersForsmallBoysSize;

						float price = smallBoysSizePrice * smallBoysBoxes;

								System.out.println("			PURCHASE DETAILS");
								System.out.println("pizza Type : Small Boys Size:");
								System.out.println(" ");

								System.out.println("  Total Boxes: "+ smallBoysBoxes);
								System.out.println("  For  "+ customersForsmallBoysSize + " person(s)\nTotal Price: #" + price + "\nLeftover Slices: "+ smallBoysSlicesLeft);
								System.out.print(" ");


								System.out.println("		Enjoy Your Pizza");

					}else{
						System.out.print("Invalid Input, Please try again");
						}
						break;



				case 3:
					System.out.print("How many customers are paying ? ");
						int customersForbigBoysSize = input.nextInt();
						System.out.println(" ");



					if(customersForbigBoysSize > 0){
						int bigBoysSize = 8;
						int bigBoysSizePrice = 3000;


						double bigBoys = (double) customersForbigBoysSize / bigBoysSize;
						double bigBoysBoxCounter = Math.ceil(bigBoys);
						int bigBoysBoxes = (int) bigBoysBoxCounter;

						int bigBoySlicesLeft = (bigBoysBoxes);

						System.out.println("		PURCHASE DETAILS");
						System.out.println("Pizza Type: Big Boys Size");
						System.out.println(" ");

						System.out.print("		Enjoy Your Pizza");
									
					}else{
						System.out.println("Invalid Input, Please Try Again");
						}
						break;


				case 4:
					System.out.print("How many customers are paying ? ");
						int customersForOdogwuSize = input.nextInt();
						System.out.println(" ");

					if(customersForOdogwuSize > 0){
						int odogwuSize = 12;
						int odogwuPrice = 5200;

							double odogwu = (double) customersForOdogwuSize / odogwuSize;
							double odogwuBoxCounter = Math.ceil(odogwu);
							int odogwuBoxes = (int) odogwuBoxCounter;

							int odogwuSlicesLeft = (odogwuBoxes*odogwuSize) - customersForOdogwuSize;
							
							float price = odogwuPrice * odogwuBoxes;
							
								System.out.println("		PURCHASE DETAILS");
								System.out.println("Pizza Type : ODOGWU Size");
								System.out.println(" ");


								System.out.println("Total Boxes: "+ odogwuBoxes);
								System.out.println("For  "+  customersForOdogwuSize  +  "person(s)\nTotal Price : #"  + price + "\nLeftover Slices : " + odogwuSlicesLeft ); 
								System.out.print(" ");
						
								System.out.print("		Enjoy Your Pizza");
			
					}else{
						System.out.print("Invalid Input, Please Try Again");					
								}
						break;

				case 0:
					System.out.print("Thank you for your patronage, See you next time");
					break;
				default:
					System.out.print("Invalid input, please enter the correct option");
					break;

				
				
				


									}
									}


									}

File: Kata/CheckOutApp/PizzaWahala/PizzaWahala.py
import math

def pizza_wahala():
    print("""
    Welcome to Iya Afeez Pizza Joint, Ajegunle.

                    MENU                         
.....................................................................................................................................................
    Pizza type            Number of Slices        Price of box        
.....................................................................................................................................................

    1. Sapa Size                 4                     2,000

    2. Small Money           6                     2,400

    3. Big Boys                  8                     3,000

    4. Odogwu                 12                    5,200
..................................................................................................................................................    
    0: Exit


    What pizza type are you buying today?
    (pick options 1-4).
    """)

    while True:
        try:
            type_chosen = int(input())

            if type_chosen == 0:
                print("Thank you for your patronage, See you next time.")
                break

            elif type_chosen == 1:
                customers_for_sapa_size = int(input("How many customers are paying? "))
                if customers_for_sapa_size > 0:
                    sapa_size = 4
                    sapa_price = 2000

                    sapa = customers_for_sapa_size / sapa_size
                    sapa_boxes = math.ceil(sapa)
                    sapa_slices_left = (sapa_boxes * sapa_size) - customers_for_sapa_size
                    total_price = sapa_price * sapa_boxes

                    print("          PURCHASE DETAILS")
                    print("Pizza Type: Sapa Size:")
                    print(f"\nTotal Boxes: {sapa_boxes}")
                    print(f"For {customers_for_sapa_size} person(s)\nTotal Price: #{total_price}\nLeftover Slices: {sapa_slices_left}")
                    print("\n     Enjoy Your Pizza")
                else:
                    print("Invalid Input, please try again")

            elif type_chosen == 2:
                customers_for_small_boys_size = int(input("How many customers are paying? "))
                if customers_for_small_boys_size > 0:
                    small_boys_size = 6
                    small_boys_size_price = 2400

                    small_boys = customers_for_small_boys_size / small_boys_size
                    small_boys_boxes = math.ceil(small_boys)
                    small_boys_slices_left = (small_boys_boxes * small_boys_size) - customers_for_small_boys_size
                    total_price = small_boys_size_price * small_boys_boxes

                    print("          PURCHASE DETAILS")
                    print("Pizza Type: Small Boys Size:")
                    print(f"\nTotal Boxes: {small_boys_boxes}")
                    print(f"For {customers_for_small_boys_size} person(s)\nTotal Price: #{total_price}\nLeftover Slices: {small_boys_slices_left}")
                    print("\n     Enjoy Your Pizza")
                else:
                    print("Invalid Input, please try again")

            elif type_chosen == 3:
                customers_for_big_boys_size = int(input("How many customers are paying? "))
                if customers_for_big_boys_size > 0:
                    big_boys_size = 8
                    big_boys_size_price = 3000

                    big_boys = customers_for_big_boys_size / big_boys_size
                    big_boys_boxes = math.ceil(big_boys)
                    big_boys_slices_left = (big_boys_boxes * big_boys_size) - customers_for_big_boys_size
                    total_price = big_boys_size_price * big_boys_boxes

                    print("          PURCHASE DETAILS")
                    print("Pizza Type: Big Boys Size:")
                    print(f"\nTotal Boxes: {big_boys_boxes}")
                    print(f"For {customers_for_big_boys_size} person(s)\nTotal Price: #{total_price}\nLeftover Slices: {big_boys_slices_left}")
                    print("\n     Enjoy Your Pizza")
                else:
                    print("Invalid Input, please try again")

            elif type_chosen == 4:
                customers_for_odogwu_size = int(input("How many customers are paying? "))
                if customers_for_odogwu_size > 0:
                    odogwu_size = 12
                    odogwu_price = 5200

                    odogwu = customers_for_odogwu_size / odogwu_size
                    odogwu_boxes = math.ceil(odogwu)
                    odogwu_slices_left = (odogwu_boxes * odogwu_size) - customers_for_odogwu_size
                    total_price = odogwu_price * odogwu_boxes

                    print("          PURCHASE DETAILS")
                    print("Pizza Type: Odogwu Size:")
                    print(f"\nTotal Boxes: {odogwu_boxes}")
                    print(f"For {customers_for_odogwu_size} person(s)\nTotal Price: #{total_price}\nLeftover Slices: {odogwu_slices_left}")
                    print("\n     Enjoy Your Pizza")
                else:
                    print("Invalid Input, please try again")

            else:
                print("Invalid input, please enter the correct option")
        except ValueError:
            print("Invalid input, please enter a valid number")

pizza_wahala()
